### PR TITLE
Controllers, DTOs and auto mapper

### PR DESCRIPTION
**Pull request by K-Karalius**
#### Controllers, DTOs:

- organized and modified DTOs
- configured an auto mapper to help map DTOs to models
- configured controller classes

#### Updated DataConcext.cs:

- The deck is now deleted together with its cards
- The state of the card is now saved as a string rather than a number

### CODE REVIEW

File: aiKart/Controllers/CardController.cs [hunk 1]
```diff
@@ -9,35 +12,158 @@ namespace aiKart.Controllers
...
+            {
+                ModelState.AddModelError("", "Something went wrong while updating a card");
+                return StatusCode(500, ModelState);
+            }
+
+            return NoContent();
+        }
+
+        [HttpPost]
+        [ProducesResponseType(200)]
```
Comment (rokas-migauskas) [inline aiKart/Controllers/CardController.cs:92]: Gal 201 created tikslingiau?

File: aiKart/Controllers/CardController.cs [hunk 2]
```diff
@@ -9,35 +12,158 @@ namespace aiKart.Controllers
...
         }
+
         [HttpGet("{cardId}")]
-        [ProducesResponseType(200, Type = typeof(Card))]
+        [ProducesResponseType(200, Type = typeof(CardDto))]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(404)]
         public IActionResult GetCard(int cardId)
         {
+            if (!_cardRepository.CardExists(cardId))
```
Comment (rokas-migauskas) [inline aiKart/Controllers/CardController.cs:30]: sitie 2 checkai visur, gal atskira metoda padaryt?

File: aiKart/Repositories/DeckRepository.cs [hunk 3]
```diff
@@ -6,51 +6,64 @@ namespace aiKart.Repositories;
...
+
+    public IEnumerable<Deck> GetDecks()
+    {
+        return _context.Decks.OrderBy(d => d.Id).ToList();
+    }
+
+    public bool DeckExistsById(int id)
     {
-        return context.Decks.OrderBy(d => d.Id).ToList();      // later this could be modified to order be by last edit
+        return _context.Decks.Any(d => d.Id == id);
```
Comment (rokas-migauskas) [inline aiKart/Repositories/DeckRepository.cs:23]: galbut cia ir fine, bet vengciau viena raide d zymet viska, deck skaitomiau butu

File: aiKart/Data/DataContext.cs [hunk 4]
```diff
@@ -1,35 +1,46 @@
...
     protected override void OnModelCreating(ModelBuilder modelBuilder)
     {
-        
+
         modelBuilder.Entity<Card>()
-            .HasKey(card => new {card.Id});
+            .HasKey(card => new { card.Id });
         modelBuilder.Entity<Deck>()
-            .HasKey(deck => new {deck.Id});
+            .HasKey(deck => new { deck.Id });
```
Comment (rokas-migauskas) [inline aiKart/Data/DataContext.cs:23]: cia new butina?

**Review by rokas-migauskas — APPROVED** [review]

File: aiKart/Controllers/CardController.cs (hunk 2 above)
Comment (rokas-migauskas) [reply aiKart/Controllers/CardController.cs:30]: eh gal nebutina, nesusitaupo tiek daug kodo

Comment (K-Karalius) [reply aiKart/Controllers/CardController.cs:30]: tas pats gausis tik kad dar papildoma funckija/metodas

File: aiKart/Data/DataContext.cs (hunk 4 above)
Comment (K-Karalius) [reply aiKart/Data/DataContext.cs:23]: cia is viso man rodos new neturetu buti, pasiziuresiu kodel cia taip

File: aiKart/Repositories/DeckRepository.cs (hunk 3 above)
Comment (K-Karalius) [reply aiKart/Repositories/DeckRepository.cs:23]: nu gal geriau butu deck rasyt o ne d, bet galvojau suprantama gan kad cia deck reiskia

**Review by MartynasJonkus — APPROVED** [review]
Looks good

**Comment by d-pilipavicius** [comment]
Everything seems OK.

File: aiKart/Controllers/CardController.cs (hunk 1 above)
Comment (K-Karalius) [reply aiKart/Controllers/CardController.cs:92]: kazkaip keistai returnint ta 201 requesta reik, nelabai supratau as jo

**Review by d-pilipavicius — COMMENTED** [review]
Everything seems OK.

**Pull request merged**